Add hasProp helper to avoid ts error

steffenkleinle opened the pull request:
### Short Description

<!-- Describe this PR in one or two sentences. -->
Add hasProp helper to avoid ts error due to typescript limitation.

### Proposed Changes

<!-- Describe this PR in more detail. -->

- Add `hasProp` to check for null/undefined props with ts acknowledging its result

### Side Effects

<!-- List all related components that have not been explicitly changed but may be affected by this PR -->

N/A

### Testing

<!-- List all things that should be tested while reviewing this PR. -->
Open the app, change a language and the city and check that everything still works.

### Resolved Issues

<!-- List all issues which should be closed when this PR is merged. -->

Fixes: N/A

---

<!--
DOR:
- [Release notes](https://github.com/digitalfabrik/integreat-app/blob/main/docs/contributing.md#release-notes) have been added for user visible changes
- Linting: `yarn lint`
- Typescript: `yarn ts:check`
- Prettier: `yarn prettier`
- Unit tests: `yarn test`
-->

LeandraH commented on shared/utils/index.ts:
```diff
@@ -45,3 +45,8 @@ export const getCategoryTiles = ({
     })
   })
+
+export const hasProp = <P extends PropertyKey, O extends { [p in P]: unknown }>(
+  obj: O,
+  p: P,
+): obj is O & { [p in P]: NonNullable<unknown> } => obj[p] !== undefined && obj[p] !== null
```
🙃  Can we call them object and prop? I don't think we really use abbreviations anywhere else, and that would make it marginally more readable, at least to me.

steffenkleinle commented:
> Hmm, I'm not getting an error when switching cities or languages. Where does this issue come from?

Yes, this is only about removing the typescript error which we suppressed with a `@ts-expect-error` so far. We introduced this in the entitlementcard at the same point, so I figured it makes sense to do this here. This was not a bug.

LeandraH commented:
> > Hmm, I'm not getting an error when switching cities or languages. Where does this issue come from?
> 
> Yes, this is only about removing the typescript error which we suppressed with a `@ts-expect-error` so far. We introduced this in the entitlementcard at the same point, so I figured it makes sense to do this here. This was not a bug, only a way to improve typesafety in our code.

Ah, nice, thanks for the explanation. So is there a `@ts-expect-error` that we can remove?

steffenkleinle commented:
> > > Hmm, I'm not getting an error when switching cities or languages. Where does this issue come from?
> > 
> > 
> > Yes, this is only about removing the typescript error which we suppressed with a `@ts-expect-error` so far. We introduced this in the entitlementcard at the same point, so I figured it makes sense to do this here. This was not a bug, only a way to improve typesafety in our code.
> 
> Ah, nice, thanks for the explanation. So is there a `@ts-expect-error` that we can remove?

See `AppContextProvider`, I already removed it:
![image](https://github.com/user-attachments/assets/ea4907a3-0780-4ea9-b06e-3636ac347089)

LeandraH commented:
> > > > Hmm, I'm not getting an error when switching cities or languages. Where does this issue come from?
> > > 
> > > 
> > > Yes, this is only about removing the typescript error which we suppressed with a `@ts-expect-error` so far. We introduced this in the entitlementcard at the same point, so I figured it makes sense to do this here. This was not a bug, only a way to improve typesafety in our code.
> > 
> > 
> > Ah, nice, thanks for the explanation. So is there a `@ts-expect-error` that we can remove?
> 
> See `AppContextProvider`, I already removed it: ![image](https://private-user-images.githubusercontent.com/29095487/421272779-ea4907a3-0780-4ea9-b06e-3636ac347089.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2ODM3MzcsIm5iZiI6MTc0MTY4MzQzNywicGF0aCI6Ii8yOTA5NTQ4Ny80MjEyNzI3NzktZWE0OTA3YTMtMDc4MC00ZWE5LWIwNmUtMzYzNmFjMzQ3MDg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzExVDA4NTcxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmNDU1NDRlMDhiODNmMmE2ZThiNzhhMTY3NzY5YWIyZmM2MTY4MTA1ZTZkYmZjZmYyNjQ4NzYzZWJjM2Q2MDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KLp3QPtVHBclWQcYhA1fTviFqM8s8OiiFgxTP8MM3tY)

Is that in a different PR? Because I don't see this change on this PR, and [the line is still there on main](https://github.com/digitalfabrik/integreat-app/blob/main/native/src/contexts/AppContextProvider.tsx#L103). Or is there a commit that you haven't pushed yet?

LeandraH reviewed (approved):
Sorry, my bad

bahaaTuffaha reviewed (approved):
LGTM .. Tested on Android Emulation (changed language + changed the area also changed the system's language)
